### ISSUE TITLE
VM has reported a failure when processing extension 'cse0'

### ISSUE THREAD

**Issue by IvanCaro** [Azure/acs-engine#1806]
**Is this a request for help?**:

---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE
---

**What version of acs-engine?**:
Version: canary
GitCommit: 8db990ba
GitTreeState: clean
---

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**
Kubernetes

**What happened**:
{
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "VMExtensionProvisioningError",
        "message": "VM has reported a failure when processing extension 'cse0'. Error message: \"Enable failed: failed to execute command: command terminated with exit status=5\n[stdout]\n\n[stderr]\nstat: cannot stat '/opt/azure/containers/provision.complete': No such file or directory\n\"."
      }

**What you expected to happen**:
{
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "VMExtensionProvisioningError",
        "message": "VM has reported a failure when processing extension 'cse0'. Error message: \"Enable failed: failed to execute command: command terminated with exit status=5\n[stdout]\n\n[stderr]\nstat: cannot stat '/opt/azure/containers/provision.complete': No such file or directory\n\"."
      }

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

**Comment by jackfrancis** [comment]
@IvanCaro could you paste the api model that you passed as input to acs-engine?

@JackQuincy I wonder if cse has an non-idiomatic execution context for this change in some cases:

https://github.com/Azure/acs-engine/commit/b5eb43b47f17d8db54ba9bfcd2a9c07bda3e6582#diff-95c1c34f292e829cdcc06906aaf5c4f1

Does it make sense that a stat failure like the above would ever short-circuit as currently implemented?

**Comment by JackQuincy** [comment]
Only reason I could see this failing is if we called set -x or something of the sort earlier in the line/command
which would be in parameters common etc.

**Comment by jackfrancis** [comment]
@JackQuincy I can reproduce consistently when specifying `"orchestratorRelease": "1.6"`. @IvanCaro Were you attempting to build a 1.6 cluster when you received this error?

**Comment by IvanCaro** [comment]
hey @jackfrancis this happened with customVnet and DNS Servers, i created the cluster without DNS (i used azure provider) and after i changed (this works).

**Comment by tamilmani1989** [comment]
@jackfrancis @JackQuincy Even I have faced this issue before and when I change the location and redeploy it worked.  This issue is not consistent and its not happening all time.

**Comment by jackfrancis** [comment]
@IvanCaro Do you agree this is an indeterminate, ephemeral issue and we should close this?

**Comment by mpluhar** [comment]
Same issue here using Kubernetes 1.6 as an orchestrator and customVnet. Is there a workaround for this?

**Comment by ducas** [comment]
I found that I was having this because my nodes (master and agents) were not able to hit the k8s.gcr.io to download kubectl. I discovered this by logging into the master and looking at /var/log/cluster-provision.log, which ended with:

    + echo 'kubernetes did not start'
    kubernetes did not start
    + exit 3

I traced this back to here - https://github.com/Azure/acs-engine/edit/master/parts/k8s/kubernetesmastercustomscript.sh#L571

This indicated it was having trouble running kubectl, so I tried invoking it from the ssh terminal. Low and behold - command not found. That file lead me to the fact that it's installed using a service called kubectl-extract. Looking at its logs using `sudo journalctl -n -u kubectl-extract` I found the following output:

    Jan 17 05:16:15 k8s-master-42756516-0 systemd[1]: Failed to start Kubectl extraction.
    Jan 17 05:16:15 k8s-master-42756516-0 systemd[1]: kubectl-extract.service: Unit entered failed state.
    Jan 17 05:16:15 k8s-master-42756516-0 systemd[1]: kubectl-extract.service: Failed with result 'exit-code'.
    Jan 17 05:16:20 k8s-master-42756516-0 systemd[1]: kubectl-extract.service: Service hold-off time over, scheduling restart.
    Jan 17 05:16:20 k8s-master-42756516-0 systemd[1]: Stopped Kubectl extraction.
    Jan 17 05:16:20 k8s-master-42756516-0 systemd[1]: Starting Kubectl extraction...
    Jan 17 05:16:35 k8s-master-42756516-0 docker[45966]: Error response from daemon: Get https://k8s-gcrio.azureedge.net/v2/hyperkube-amd64/manifests/v1.7.9: Get https://k8s.gcr.io/v2/token?scope=repository%3Ahyperkube-amd64%3Apull&service=k8s.gcr.io: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
    Jan 17 05:16:35 k8s-master-42756516-0 systemd[1]: kubectl-extract.service: Control process exited, code=exited status=1

So there was a problem downloading kubectl from k8s.gcr.io. Turns out it was a DNS problem, but that's just my network... Hope this helps someone debug a related issue.

**Comment by jsturtevant** [comment]
I ran into this as well.  It failed when ran in westus2. I changed to eastus and it worked

**Comment by nakahparis** [comment]
I'm having same issues when deploying to custom VNET on WestEurope using acs-engine 0.12.4 and Kubernetes 1.9.1.
I've backuped all logs from /var/log/azure & /var/log/containers if it can help.

**Comment by msorby** [comment]
Just to acknowledge this. I get the exact same thing, custom VNET in West Europe, North Europe works just fine.

**Comment by rodrigoffonseca** [comment]
I got the same error and the problem was the DNS resolution at the VNET. After I fixed my custom DNS servers to resolve internet names everything worked fine.

**Comment by jackfrancis** [comment]
cse* errors generally are a result of the provisioning process on the host failing. I'd like to keep this ticket open to encourage folks to share (bad) experiences. We're working on (1) improving logging around this and (2) hunting down transient errors (such as lack of DNS access would incur) and try, where appropriate, to introduce add'l resilience.

**Comment by feiskyer** [comment]
also met same problem in eastus. cse0 timed out and the master VM can't ssh

**Comment by jackfrancis** [comment]
Just identified one cause of transient cse errors (DNS availability race condition on cluster provisioning), added some retry resiliency and am hoping that eliminates that symptom. @feiskyer please try to repro using master next week and let me know if you can, thanks!

**Comment by feiskyer** [comment]
@jackfrancis sure

**Comment by ilyalukyanov** [comment]
@jackfrancis Thanks for pushing the fix. I've tested with the latest master, but unfortunately it didn't help in my case.

I consistently face this issue if networking is set to `azure`. However, even before your fixes I was able to deploy at least several times in a row with networking set to `none` (for some reason networking was extremely unreliable so that's not a solution) and haven't had it failing.

In case this is useful, here's my template:
```
{
    "apiVersion": "vlabs",
    "properties": {
        "orchestratorProfile": {
            "orchestratorType": "Kubernetes",
            "orchestratorRelease": "1.9",
            "orchestratorVersion": "1.9.2",
            "kubernetesConfig": {
                "networkPolicy": "azure"
            }
        },
        "masterProfile": {
            "count": 1,
            "dnsPrefix": "z-loyalty-development-k8s",
            "vnetSubnetId": "<value>",
            "firstConsecutiveStaticIP": "172.19.5.100",
            "vmSize": "Standard_D2_v2"
        },
        "agentPoolProfiles": [{
            "availabilityProfile": "AvailabilitySet",
            "count": 2,
            "name": "pool1",
            "OSDiskSizeGB": 400,
            "storageProfile" : "ManagedDisks",
            "vmSize": "Standard_D2_v2",
            "vnetSubnetId": "<value>",
            "osType": "Linux",
            "distro": "ubuntu"
         }],
        "linuxProfile": {
            "adminUsername": "<value>",
            "ssh": {
                "publicKeys": [{ "keyData": "<value>" }]
            },
            "secrets": []
        },
        "servicePrincipalProfile": {
            "clientId": "<value>",
            "secret": "<value>"
        },
        "certificateProfile": {}
    }
}
```

And my `acs-engine version` output:
```
Version: canary
GitCommit: 7923b960
GitTreeState: clean
```

Trying with calico, will keep this thread updated.

**Comment by msorby** [comment]
I've been testing a lot to day with ace-engine 0.12.5 and I've yet to run into this issue. Last week with 0.12.4 and 0.12.2 I got it all the time. So it seems to be much better 👍

**Comment by jackfrancis** [comment]
@ilyalukyanov This PR also moves the ball forward:

https://github.com/Azure/acs-engine/pull/2196

That is aimed for master today, should reduce further cse flakiness.

**Comment by msorby** [comment]
I've literally deployed 20 times today without issues. Then suddenly the extension error popped up again. For the exact same generated template. This was a template generated with acs-engine 0.12.5.
So there is still some flakiness left ;-)

**Comment by ilyalukyanov** [comment]
@jackfrancis thanks for prompt fixes! I'll give them a go later this week and will update this thread.

**Comment by idanshahar** [comment]
This is still happening in West Europe.
I'm using acs-engine 0.12.5

**Comment by Jarlotee** [comment]
Is there a work around to get the partial deployment into a health state?

**Comment by jackfrancis** [comment]
@idanshahar Are you able to build from master? Much of the work post v0.12 has been identifying transient issues with provision scripts (and dependencies), which is where CSE deployment errors originate.

@Jarlotee Depending on the scenario you could cherry-pick through the provision script `/opt/azure/containers/provision.sh` and manually execute the failed commands, but that would be tedious for a new cluster. The easiest path forward is to re-build, again using built-from-master binary, if possible.

Thanks for your endurance, all. :)

**Comment by Jarlotee** [comment]
@jackfrancis and anyone else who gets bitten by this.

My issue turned out to be in the SPN password with a % in it!

The password was truncated at the % which caused the subsequent failure.

**Comment by jackfrancis** [comment]
Ugh. See https://github.com/Azure/acs-engine/issues/1208

We'll prioritize this in the next release cycle. Thanks for sharing @Jarlotee !

**Comment by idanshahar** [comment]
@jackfrancis Yes I can do so, but still need a patch for a customer. when does the next version suppose to be released? BTW there is another issue exists in the master branch... #2198

**Comment by jackfrancis** [comment]
ETA for the next release is this week. Issue #2198 is a long-standing gap for Windows clusters. This won't be addressed in the next release.

**Comment by robsonpeixoto** [comment]
I got the same problem:

Version:
```
~/go/src/github.com/Azure/acs-engine/bin/acs-engine version
Version: canary
GitCommit: e3b22c48
GitTreeState: clean
```

API Model:
```
{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.8",
      "kubernetesConfig": {
        "networkPolicy": "calico"
      }
    },
    "masterProfile": {
      "count": 1,
      "dnsPrefix": "homolog",
      "vmSize": "Standard_D2_v3",
      "vnetSubnetId": "<SAME_VNET_FOR_MASTER_AND_AGENTS>",
      "firstConsecutiveStaticIP": "172.31.0.10",
      "vnetCidr": "172.31.0.6/26",
      "distro": "ubuntu"
    },
    "agentPoolProfiles": [
      {
        "name": "agentpool",
        "count": 2,
        "vmSize": "Standard_D2_v3",
        "availabilityProfile": "AvailabilitySet",
        "vnetSubnetId": "<SAME_VNET_FOR_MASTER_AND_AGENTS>",
        "distro": "ubuntu"
      }
    ],
    "linuxProfile": {
      "adminUsername": "<VALUE>",
      "ssh": {
        "publicKeys": [
          {
            "keyData": "<VALUE>"
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "clientId": "<value>",
      "secret": "<value>"
    }
  }
}

```

**Comment by jackfrancis** [comment]
@robsonpeixoto Could you paste the last ten or so lines of `/var/log/azure/cluster-provision.log` and `/var/log/cloud-init-output.log` on the master?

**Comment by robsonpeixoto** [comment]
@jackfrancis The problem is not happing again. I deleted the `_output` directory to create another cluster and when I try again, the error not happens anymore.

**Comment by richardjharding** [comment]
I'm seeing this consistently with 

```
Version: v0.13.0
GitCommit: 4d00356a
GitTreeState: clean
```

Error

```
Deployment failed. Correlation ID: d88dd621-d41a-404c-aaa3-8ce38c9615a6. {
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "VMExtensionProvisioningError",
        "message": "VM has reported a failure when processing extension 'cse0'. Error message: \"Enable failed: failed to execute command: command terminated with exit status=3\n[stdout]\n\n[stderr]\n\"."
      }
    ]
  }
}
```
with following versions of k8s and using custom vnet
```
"orchestratorRelease": "1.9",
"orchestratorVersion": "1.9.3",
```

**Comment by jackfrancis** [comment]
@richardjharding Thanks very much for sharing. Would you be willing/able to try and repro from the code at this PR?:

https://github.com/Azure/acs-engine/pull/2225

The changes there add some more checks around "is the cluster ready" to account for possible race conditions in parallel operations. I'm not sure we're committed to merging it, as it may add delays to deployment time, but if we get convincing evidence that it removes edge case failure behaviors we'll consider that strongly.

**Comment by richardjharding** [comment]
sure I'll give that a go over the next couple of days or so and report back

**Comment by richardjharding** [comment]
@jackfrancis  I should have checked - is that change compatible with the 0.13 release? - I need the AAD admin group object id that was added in that release

**Comment by jackfrancis** [comment]
@richardjharding yep

**Comment by richardjharding** [comment]
@jackfrancis  - I built and tested #2225 - it ran for a lot longer but came back the same error I'm afraid

```
Deployment failed. Correlation ID: 80a7b519-45a9-44ce-be50-e294e095aa32. {
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "VMExtensionProvisioningError",
        "message": "VM has reported a failure when processing extension 'cse0'. Error message: \"Enable failed: failed to execute command: command terminated with exit status=3\n[stdout]\n\n[stderr]\n\"."
      }
    ]
  }
}
```

**Comment by EPinci** [comment]
Same in here, getting:
`        "code": "VMExtensionProvisioningError",
        "message": "VM has reported a failure when processing extension 'cse0'. Error message: \"Enable failed: failed to execute command: command terminated with exit status=3\n[stdout]\n\n[stderr]\n\"."`
trying to deploy on WestEU. Any workaround?

**Comment by jackfrancis** [comment]
@EPinci which version of acs-engine did you deploy with?

**Comment by EPinci** [comment]
@jackfrancis I used 0.13.0 but in my case this was a weird network routing issue (couldn't get the nodes to the repo to pull the components from).

**Comment by ravibarkhani** [comment]
Hello,

I am getting same error, tried in both north ans south UK location. With one master and one agent it works fine, but when I choose 3 master and 3 agent it start failing extensions to agents are perfectly installed. Deploying in custom VNET with azure provided DNS. I checked from master node it is able to connect to internet(nslookup docker.io gives me proper response). In my case I am using CIS hardened Linux image. With acs-engine default OS ubuntu it works fine.

Deployment failed. Correlation ID: 3750b0a7-c1ac-46ea-a885-7bff089ab22c. {
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "VMExtensionProvisioningError",
        "message": "VM has reported a failure when processing extension 'cse1'. Error message: \"Enable failed: failed to execute command: command terminated with exit status=5\n[stdout]\n\n[stderr]\nstat: cannot stat '/opt/azure/containers/provision.complete': No such file or directory\n\"."
      }
    ]
  }
}

==acs-engine version ==
Version: v0.12.5
GitCommit: 9fa6a690
GitTreeState: clean


In the journalctl logs I see error : ****publish** error: etcdserver: request timed out**

**Comment by jackfrancis** [comment]
Thanks for reporting @ravibarkhani, that error should not occur on a more recent build of acs-engine. (We've introduced lots of I/O-dependent retries + race condition resiliency during cluster deployment since v0.12.5)

**Comment by ravibarkhani** [comment]
With the version 13.1 extension deployment timed is timed out after 1 hour 30 min. I am running k*s versiion 1.8 and version 1.8.7. Once will try with new version of k8s and share the details.

**Comment by khaldoune** [comment]
@jackfrancis @yastij @jcorioland 
I'm getting the same error with acs **0.14**, NorthEurope, kubernetes 1.9.3 and kubernetes 1.9.4.
I'll send the logs from /var/log/cloud-init-output.log once I have access to the VMs.

**Comment by idanshahar** [comment]
@khaldoune 
can you share your API model?

**Comment by khaldoune** [comment]
@idanshahar 
Here you go:
`{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.9",
      "orchestratorVersion": "1.9.4",
      "kubernetesConfig": {
        "kubernetesImageBase": "k8s-gcrio.azureedge.net/",
        "clusterSubnet": "10.10.0.0/16",
        "dnsServiceIP": "192.168.1.10",
        "serviceCidr": "192.168.1.0/24",
        "networkPolicy": "calico",
        "maxPods": 110,
        "dockerBridgeSubnet": "172.17.0.1/16",
        "useInstanceMetadata": true,
        "enableRbac": true,
        "enableSecureKubelet": true,
        "enableAggregatedAPIs": true,
        "privateCluster": {
          "enabled": false
        },
        "gchighthreshold": 85,
        "gclowthreshold": 80,
        "etcdVersion": "3.2.16",
        "etcdDiskSizeGB": "16",
        "enablePodSecurityPolicy": true,
        "addons": [
          {
            "name": "tiller",
            "enabled": false
          },
          {
            "name": "aci-connector",
            "enabled": false,
            "containers": [
              {
                "name": "aci-connector",
                "cpuRequests": "50m",
                "memoryRequests": "150Mi",
                "cpuLimits": "50m",
                "memoryLimits": "150Mi"
              }
            ],
            "config": {
              "nodeName": "aci-connector",
              "os": "Linux",
              "region": "westus",
              "taint": "azure.com/aci"
            }
          },
          {
            "name": "kubernetes-dashboard",
            "enabled": true,
            "containers": [
              {
                "name": "kubernetes-dashboard",
                "cpuRequests": "300m",
                "memoryRequests": "150Mi",
                "cpuLimits": "300m",
                "memoryLimits": "150Mi"
              }
            ]
          },
          {
            "name": "rescheduler",
            "enabled": false,
            "containers": [
              {
                "name": "rescheduler",
                "cpuRequests": "10m",
                "memoryRequests": "100Mi",
                "cpuLimits": "10m",
                "memoryLimits": "100Mi"
              }
            ]
          },
          {
            "name": "metrics-server",
            "enabled": true,
            "containers": [
              {
                "name": "metrics-server"
              }
            ]
          }
        ],
        "kubeletConfig": {
          "--address": "0.0.0.0",
          "--allow-privileged": "true",
          "--anonymous-auth": "false",
          "--authentication-token-webhook": "true",
          "--authorization-mode": "Webhook",
          "--azure-container-registry-config": "/etc/kubernetes/azure.json",
          "--cadvisor-port": "0",
          "--cgroups-per-qos": "true",
          "--client-ca-file": "/etc/kubernetes/certs/ca.crt",
          "--cloud-config": "/etc/kubernetes/azure.json",
          "--cloud-provider": "azure",
          "--cluster-dns": "192.168.1.10",
          "--cluster-domain": "cluster.local",
          "--enforce-node-allocatable": "pods",
          "--event-qps": "0",
          "--eviction-hard": "memory.available<100Mi,nodefs.available<10%,nodefs.inodesFree<5%",
          "--feature-gates": "",
          "--image-gc-high-threshold": "85",
          "--image-gc-low-threshold": "80",
          "--keep-terminated-pod-volumes": "false",
          "--kubeconfig": "/var/lib/kubelet/kubeconfig",
          "--max-pods": "110",
          "--network-plugin": "cni",
          "--node-status-update-frequency": "10s",
          "--non-masquerade-cidr": "10.10.0.0/16",
          "--pod-infra-container-image": "k8s-gcrio.azureedge.net/pause-amd64:3.1",
          "--pod-manifest-path": "/etc/kubernetes/manifests"
        },
        "controllerManagerConfig": {
          "--address": "0.0.0.0",
          "--allocate-node-cidrs": "true",
          "--cloud-config": "/etc/kubernetes/azure.json",
          "--cloud-provider": "azure",
          "--cluster-cidr": "10.10.0.0/16",
          "--cluster-name": "k8s-noprd",
          "--cluster-signing-cert-file": "/etc/kubernetes/certs/ca.crt",
          "--cluster-signing-key-file": "/etc/kubernetes/certs/ca.key",
          "--feature-gates": "ServiceNodeExclusion=true",
          "--horizontal-pod-autoscaler-use-rest-clients": "true",
          "--kubeconfig": "/var/lib/kubelet/kubeconfig",
          "--leader-elect": "true",
          "--node-cidr-mask-size": "27",
          "--node-monitor-grace-period": "40s",
          "--pod-eviction-timeout": "60s",
          "--profiling": "false",
          "--root-ca-file": "/etc/kubernetes/certs/ca.crt",
          "--route-reconciliation-period": "10s",
          "--service-account-private-key-file": "/etc/kubernetes/certs/apiserver.key",
          "--terminated-pod-gc-threshold": "100",
          "--use-service-account-credentials": "true",
          "--v": "2"
        },
        "cloudControllerManagerConfig": {
          "--allocate-node-cidrs": "true",
          "--cloud-config": "/etc/kubernetes/azure.json",
          "--cloud-provider": "azure",
          "--cluster-cidr": "10.10.0.0/16",
          "--cluster-name": "k8s-noprd",
          "--kubeconfig": "/var/lib/kubelet/kubeconfig",
          "--leader-elect": "true",
          "--profiling": "false",
          "--route-reconciliation-period": "10s",
          "--v": "2"
        },
        "apiServerConfig": {
          "--admission-control": "NamespaceLifecycle,LimitRanger,ServiceAccount,DefaultStorageClass,ResourceQuota,DenyEscalatingExec,AlwaysPullImages,PodSecurityPolicy",
          "--advertise-address": "<kubernetesAPIServerIP>",
          "--allow-privileged": "true",
          "--anonymous-auth": "false",
          "--audit-log-maxage": "30",
          "--audit-log-maxbackup": "10",
          "--audit-log-maxsize": "100",
          "--audit-log-path": "/var/log/audit.log",
          "--audit-policy-file": "/etc/kubernetes/manifests/audit-policy.yaml",
          "--authorization-mode": "Node,RBAC",
          "--bind-address": "0.0.0.0",
          "--client-ca-file": "/etc/kubernetes/certs/ca.crt",
          "--cloud-config": "/etc/kubernetes/azure.json",
          "--cloud-provider": "azure",
          "--endpoint-reconciler-type": "lease",
          "--etcd-cafile": "/etc/kubernetes/certs/ca.crt",
          "--etcd-certfile": "/etc/kubernetes/certs/etcdclient.crt",
          "--etcd-keyfile": "/etc/kubernetes/certs/etcdclient.key",
          "--etcd-quorum-read": "true",
          "--etcd-servers": "https://127.0.0.1:2379",
          "--insecure-port": "8080",
          "--kubelet-client-certificate": "/etc/kubernetes/certs/client.crt",
          "--kubelet-client-key": "/etc/kubernetes/certs/client.key",
          "--profiling": "false",
          "--proxy-client-cert-file": "/etc/kubernetes/certs/proxy.crt",
          "--proxy-client-key-file": "/etc/kubernetes/certs/proxy.key",
          "--repair-malformed-updates": "false",
          "--requestheader-allowed-names": "",
          "--requestheader-client-ca-file": "/etc/kubernetes/certs/proxy-ca.crt",
          "--requestheader-extra-headers-prefix": "X-Remote-Extra-",
          "--requestheader-group-headers": "X-Remote-Group",
          "--requestheader-username-headers": "X-Remote-User",
          "--secure-port": "443",
          "--service-account-key-file": "/etc/kubernetes/certs/apiserver.key",
          "--service-account-lookup": "true",
          "--service-cluster-ip-range": "192.168.1.0/24",
          "--storage-backend": "etcd3",
          "--tls-cert-file": "/etc/kubernetes/certs/apiserver.crt",
          "--tls-private-key-file": "/etc/kubernetes/certs/apiserver.key",
          "--v": "4"
        }
      }
    },
    "masterProfile": {
      "count": 5,
      "dnsPrefix": "k8s-noprd",
      "vmSize": "Standard_D2_v2",
      "osDiskSizeGB": 80,
      "vnetSubnetID": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/master_subnet",
      "firstConsecutiveStaticIP": "198.18.187.230",
      "storageProfile": "ManagedDisks",
      "oauthEnabled": false,
      "preProvisionExtension": {
        "name": "setup",
        "singleOrAll": "",
        "template": ""
      },
      "extensions": [],
      "distro": "ubuntu",
      "kubernetesConfig": {
        "kubeletConfig": {
          "--address": "0.0.0.0",
          "--allow-privileged": "true",
          "--anonymous-auth": "false",
          "--authentication-token-webhook": "true",
          "--authorization-mode": "Webhook",
          "--azure-container-registry-config": "/etc/kubernetes/azure.json",
          "--cadvisor-port": "0",
          "--cgroups-per-qos": "true",
          "--client-ca-file": "/etc/kubernetes/certs/ca.crt",
          "--cloud-config": "/etc/kubernetes/azure.json",
          "--cloud-provider": "azure",
          "--cluster-dns": "192.168.1.10",
          "--cluster-domain": "cluster.local",
          "--enforce-node-allocatable": "pods",
          "--event-qps": "0",
          "--eviction-hard": "memory.available<100Mi,nodefs.available<10%,nodefs.inodesFree<5%",
          "--feature-gates": "",
          "--image-gc-high-threshold": "85",
          "--image-gc-low-threshold": "80",
          "--keep-terminated-pod-volumes": "false",
          "--kubeconfig": "/var/lib/kubelet/kubeconfig",
          "--max-pods": "110",
          "--network-plugin": "cni",
          "--node-status-update-frequency": "10s",
          "--non-masquerade-cidr": "10.10.0.0/16",
          "--pod-infra-container-image": "k8s-gcrio.azureedge.net/pause-amd64:3.1",
          "--pod-manifest-path": "/etc/kubernetes/manifests"
        }
      }
    },
    "agentPoolProfiles": [
      {
        "name": "workers0",
        "count": 5,
        "vmSize": "Standard_D2_v2",
        "osDiskSizeGB": 80,
        "osType": "Linux",
        "availabilityProfile": "AvailabilitySet",
        "storageProfile": "ManagedDisks",
        "diskSizesGB": [
          50
        ],
        "vnetSubnetID": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/worker_subnet",
        "distro": "ubuntu",
        "kubernetesConfig": {
          "kubeletConfig": {
            "--address": "0.0.0.0",
            "--allow-privileged": "true",
            "--anonymous-auth": "false",
            "--authentication-token-webhook": "true",
            "--authorization-mode": "Webhook",
            "--azure-container-registry-config": "/etc/kubernetes/azure.json",
            "--cadvisor-port": "0",
            "--cgroups-per-qos": "true",
            "--client-ca-file": "/etc/kubernetes/certs/ca.crt",
            "--cloud-config": "/etc/kubernetes/azure.json",
            "--cloud-provider": "azure",
            "--cluster-dns": "192.168.1.10",
            "--cluster-domain": "cluster.local",
            "--enforce-node-allocatable": "pods",
            "--event-qps": "0",
            "--eviction-hard": "memory.available<100Mi,nodefs.available<10%,nodefs.inodesFree<5%",
            "--feature-gates": "Accelerators=true",
            "--image-gc-high-threshold": "85",
            "--image-gc-low-threshold": "80",
            "--keep-terminated-pod-volumes": "false",
            "--kubeconfig": "/var/lib/kubelet/kubeconfig",
            "--max-pods": "110",
            "--network-plugin": "cni",
            "--node-status-update-frequency": "10s",
            "--non-masquerade-cidr": "10.10.0.0/16",
            "--pod-infra-container-image": "k8s-gcrio.azureedge.net/pause-amd64:3.1",
            "--pod-manifest-path": "/etc/kubernetes/manifests"
          }
        },
        "fqdn": "",
        "preProvisionExtension": {
          "name": "setup_node",
          "singleOrAll": "",
          "template": ""
        },
        "extensions": []
      }
    ],
    "linuxProfile": {
      "adminUsername": "k8s",
      "ssh": {
        "publicKeys": [
          {
            "keyData": "xxxxxxxxxxxxxxxxxxxxxxxxxx"
          }
        ]
      }
    },
    "extensionProfiles": [
      {
        "name": "setup_node",
        "version": "v1",
        "extensionParameters": "198.18.192.4 k8s-noprd.xpod.carrefour.com",
        "rootURL": "https://gitlab.com/octo-carrefour-k8s/acs-extensions/raw/master/",
        "script": "setup.sh"
      },
      {
        "name": "setup",
        "version": "v1",
        "extensionParameters": "198.18.192.4 k8s-noprd.xpod.carrefour.com",
        "rootURL": "https://gitlab.com/octo-carrefour-k8s/acs-extensions/raw/master/",
        "script": "setup.sh"
      }
    ],
    "servicePrincipalProfile": {
      "clientId": "fc24d5cf-73aa-439a-bc89-7be37baba0f5",
      "secret": "BX3A75IjAwqh+Per233s4jJXeSYVejv1Wqur9hkdln4="
    },
    "certificateProfile": {
      "caCertificate": "xxxxxxxxxxxxxxxxxxxxx",
      "caPrivateKey": "xxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "apiServerCertificate": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "apiServerPrivateKey": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "clientCertificate": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "clientPrivateKey": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "kubeConfigCertificate": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "kubeConfigPrivateKey": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "etcdServerCertificate": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "etcdServerPrivateKey": "xxxxxxxxxxxxxxxxxxxxxxxxxx",
      "etcdClientCertificate": "xxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "etcdClientPrivateKey": "xxxxxxxxxxxxxxxxxxxxxxxx",
      "etcdPeerCertificates": [
        "xxxxxxxxxxxxxxxxxxxxxxxx",
        "xxxxxxxxxxxxxxx",
        "xxxxxxxxxxxxx",
        "xxxxxxxxxxxxxxxxx",
        "xxxxxxxxxxxxxxx"
      ],
      "etcdPeerPrivateKeys": [
        "xxxxx",
        "xxxxxxxxxxxxxx",
        "xxxxxxxxxxxxxxxxxx",
        "xxxxxxxxxxxxxxxxx",
        "xxxxxxxxxxxxxxxxxx"
      ]
    }
  }
}
`
@jackfrancis cc @yastij cc @jcorioland

Thanks.

**Comment by khaldoune** [comment]
@idanshahar Did you manage to reproduce this bug?
This problem does not occures with acs 0.13, it's a regression (@jackfrancis ) from my point of view.
Cc @yastij cc @jcorioland

**Comment by khaldoune** [comment]
@jackfrancis yes absolutely, if you revert just the acs binaries to 0.13 and the kubernetes version to 1.9.3 (acs 0.13 does not support k8s 1.9.4), in the apimodel that I have provided, this issue will not occures. 
How can I help?

**Comment by idanshahar** [comment]
@khaldoune Not yet.
First, let's identify if the "cse" error came from the master or the agents.
To do so, go to your cluster resource group -> deployments, look for your Kubernetes deployment name and click on it, you can find there the stage that it failed on.

**Comment by khaldoune** [comment]
@idanshahar cc @jackfrancis
The error occures on all masters.

**Comment by idanshahar** [comment]
@khaldoune 
Can you ssh to the master and check if kubelet is running?
if kubelet isn't running, try run kubelet and past the output here.

**Comment by khaldoune** [comment]
@idanshahar 
`$ sudo systemctl status kubelet
● kubelet.service - Kubelet
   Loaded: loaded (/etc/systemd/system/kubelet.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2018-03-19 17:55:01 UTC; 2 days ago
 Main PID: 7373 (docker)
    Tasks: 8
   Memory: 6.4M
      CPU: 11.357s
   CGroup: /system.slice/kubelet.service
           └─7373 /usr/bin/docker run --net=host --pid=host --privileged --rm --volume=/dev:/dev --volume=/sys:/sys:ro --volume=/var/run:/var

Mar 21 22:16:42 k8s-master-35332392-0 docker[7373]: E0321 22:16:42.928108    7928 kubelet.go:2104] Container runtime network not ready: Netwo
Mar 21 22:16:45 k8s-master-35332392-0 docker[7373]: E0321 22:16:45.893950    7928 kubelet.go:1606] Failed creating a mirror pod for "kube-add
Mar 21 22:16:47 k8s-master-35332392-0 docker[7373]: W0321 22:16:47.928736    7928 cni.go:171] Unable to update cni config: No networks found
Mar 21 22:16:47 k8s-master-35332392-0 docker[7373]: E0321 22:16:47.928971    7928 kubelet.go:2104] Container runtime network not ready: Netwo
Mar 21 22:16:47 k8s-master-35332392-0 docker[7373]: I0321 22:16:47.951090    7928 kubelet_node_status.go:497] Using Node Hostname from cloudp
Mar 21 22:16:52 k8s-master-35332392-0 docker[7373]: W0321 22:16:52.929995    7928 cni.go:171] Unable to update cni config: No networks found
Mar 21 22:16:52 k8s-master-35332392-0 docker[7373]: E0321 22:16:52.930176    7928 kubelet.go:2104] Container runtime network not ready: Netwo
Mar 21 22:16:57 k8s-master-35332392-0 docker[7373]: W0321 22:16:57.930954    7928 cni.go:171] Unable to update cni config: No networks found
Mar 21 22:16:57 k8s-master-35332392-0 docker[7373]: E0321 22:16:57.931926    7928 kubelet.go:2104] Container runtime network not ready: Netwo
Mar 21 22:16:57 k8s-master-35332392-0 docker[7373]: I0321 22:16:57.997106    7928 kubelet_node_status.go:497] Using Node Hostname from cloudp`

**Comment by khaldoune** [comment]
@idanshahar 
`journalctl -u kubelet
Hint: You are currently not seeing messages from other users and the system.
      Users in the 'systemd-journal' group can see all messages. Pass -q to
      turn off this notice.
-- No entries --
`

**Comment by idanshahar** [comment]
@khaldoune 
'journalctl -u kubelet' will output the kubelet logs.
Try run this too and upload the output.
Cc @jackfrancis

**Comment by khaldoune** [comment]
@idanshahar 
`k8s@k8s-master-35332392-0:~$ docker ps | grep kubelet

k8s@k8s-master-35332392-0:~$
`

The service status is active but it seems like there is no kubelet container running.

`sudo systemctl restart kubelet` 
`$ sudo systemctl status kubelet
● kubelet.service - Kubelet
   Loaded: loaded (/etc/systemd/system/kubelet.service; enabled; vendor preset: enabled)
   Active: active (running) since Wed 2018-03-21 22:30:56 UTC; 23s ago
  Process: 36504 ExecStartPre=/sbin/iptables -t nat --list (code=exited, status=0/SUCCESS)
  Process: 36502 ExecStartPre=/sbin/ebtables -t nat --list (code=exited, status=0/SUCCESS)
  Process: 36496 ExecStartPre=/sbin/sysctl -w net.ipv4.tcp_retries2=8 (code=exited, status=0/SUCCESS)
  Process: 36494 ExecStartPre=/bin/mount --make-shared /var/lib/kubelet (code=exited, status=0/SUCCESS)
  Process: 36485 ExecStartPre=/bin/bash -c if [ $(mount | grep "/var/lib/kubelet" | wc -l) -le 0 ] ; then /bin/mount --bind /var/lib/kubelet
  Process: 36482 ExecStartPre=/bin/mkdir -p /var/lib/cni (code=exited, status=0/SUCCESS)
  Process: 36478 ExecStartPre=/bin/mkdir -p /var/lib/kubelet (code=exited, status=0/SUCCESS)
  Process: 36434 ExecStartPre=/bin/bash /opt/azure/containers/kubelet.sh (code=exited, status=0/SUCCESS)
 Main PID: 36510 (docker)
    Tasks: 7
   Memory: 5.4M
      CPU: 67ms
   CGroup: /system.slice/kubelet.service
           └─36510 /usr/bin/docker run --net=host --pid=host --privileged --rm --volume=/dev:/dev --volume=/sys:/sys:ro --volume=/var/run:/va

Mar 21 22:31:05 k8s-master-35332392-0 docker[36510]: W0321 22:31:05.310799   36534 cni.go:171] Unable to update cni config: No networks found
Mar 21 22:31:05 k8s-master-35332392-0 docker[36510]: E0321 22:31:05.310943   36534 kubelet.go:2104] Container runtime network not ready: Netw
Mar 21 22:31:08 k8s-master-35332392-0 docker[36510]: I0321 22:31:08.189305   36534 kubelet_node_status.go:497] Using Node Hostname from cloud
Mar 21 22:31:10 k8s-master-35332392-0 docker[36510]: W0321 22:31:10.312301   36534 cni.go:171] Unable to update cni config: No networks found
Mar 21 22:31:10 k8s-master-35332392-0 docker[36510]: E0321 22:31:10.312418   36534 kubelet.go:2104] Container runtime network not ready: Netw
Mar 21 22:31:15 k8s-master-35332392-0 docker[36510]: W0321 22:31:15.313228   36534 cni.go:171] Unable to update cni config: No networks found
Mar 21 22:31:15 k8s-master-35332392-0 docker[36510]: E0321 22:31:15.313397   36534 kubelet.go:2104] Container runtime network not ready: Netw
Mar 21 22:31:18 k8s-master-35332392-0 docker[36510]: I0321 22:31:18.244506   36534 kubelet_node_status.go:497] Using Node Hostname from cloud
Mar 21 22:31:20 k8s-master-35332392-0 docker[36510]: W0321 22:31:20.314319   36534 cni.go:171] Unable to update cni config: No networks found
Mar 21 22:31:20 k8s-master-35332392-0 docker[36510]: E0321 22:31:20.314692   36534 kubelet.go:2104] Container runtime network not ready: Netw`

Docker ps | grep kubelet => nothing

ps -ef | grep kubelet =>
`root      36510      1  0 22:30 ?        00:00:00 /usr/bin/docker run --net=host --pid=host --privileged --rm --volume=/dev:/dev --volume=/sys:/sys:ro --volume=/var/run:/var/run:rw --volume=/var/lib/cni/:/var/lib/cni:rw --volume=/sbin/apparmor_parser/:/sbin/apparmor_parser:rw --volume=/var/lib/docker/:/var/lib/docker:rw --volume=/var/lib/containers/:/var/lib/containers:rw --volume=/var/lib/kubelet/:/var/lib/kubelet:shared --volume=/var/log:/var/log:rw --volume=/etc/kubernetes/:/etc/kubernetes:ro --volume=/srv/kubernetes/:/srv/kubernetes:ro --volume=/etc/cni/:/etc/cni:ro --volume=/opt/cni/:/opt/cni:ro --volume=/var/lib/waagent/ManagedIdentity-Settings:/var/lib/waagent/ManagedIdentity-Settings:ro --volume=/etc/kubernetes/volumeplugins:/etc/kubernetes/volumeplugins:rw k8s-gcrio.azureedge.net/hyperkube-amd64:v1.9.3 /hyperkube kubelet --enable-server --node-labels=kubernetes.io/role=master,kubernetes.azure.com/cluster=apps-k8s-noprd-rg --v=2 --non-masquerade-cidr=10.10.0.0/16 --volume-plugin-dir=/etc/kubernetes/volumeplugins --address=0.0.0.0 --allow-privileged=true --anonymous-auth=false --authentication-token-webhook=true --authorization-mode=Webhook --azure-container-registry-config=/etc/kubernetes/azure.json --cadvisor-port=0 --cgroups-per-qos=true --client-ca-file=/etc/kubernetes/certs/ca.crt --cloud-config=/etc/kubernetes/azure.json --cloud-provider=azure --cluster-dns=192.168.1.10 --cluster-domain=cluster.local --enforce-node-allocatable=pods --event-qps=0 --eviction-hard=memory.available<100Mi,nodefs.available<10%,nodefs.inodesFree<5% --feature-gates= --image-gc-high-threshold=85 --image-gc-low-threshold=80 --keep-terminated-pod-volumes=false --kubeconfig=/var/lib/kubelet/kubeconfig --max-pods=110 --network-plugin=cni --node-status-update-frequency=10s --non-masquerade-cidr=10.10.0.0/16 --pod-infra-container-image=k8s-gcrio.azureedge.net/pause-amd64:3.1 --pod-manifest-path=/etc/kubernetes/manifests --register-node=true --register-with-taints=node-role.kubernetes.io/master=true:NoSchedule
root      36534  36518  3 22:30 ?        00:00:03 /hyperkube kubelet --enable-server --node-labels=kubernetes.io/role=master,kubernetes.azure.com/cluster=apps-k8s-noprd-rg --v=2 --non-masquerade-cidr=10.10.0.0/16 --volume-plugin-dir=/etc/kubernetes/volumeplugins --address=0.0.0.0 --allow-privileged=true --anonymous-auth=false --authentication-token-webhook=true --authorization-mode=Webhook --azure-container-registry-config=/etc/kubernetes/azure.json --cadvisor-port=0 --cgroups-per-qos=true --client-ca-file=/etc/kubernetes/certs/ca.crt --cloud-config=/etc/kubernetes/azure.json --cloud-provider=azure --cluster-dns=192.168.1.10 --cluster-domain=cluster.local --enforce-node-allocatable=pods --event-qps=0 --eviction-hard=memory.available<100Mi,nodefs.available<10%,nodefs.inodesFree<5% --feature-gates= --image-gc-high-threshold=85 --image-gc-low-threshold=80 --keep-terminated-pod-volumes=false --kubeconfig=/var/lib/kubelet/kubeconfig --max-pods=110 --network-plugin=cni --node-status-update-frequency=10s --non-masquerade-cidr=10.10.0.0/16 --pod-infra-container-image=k8s-gcrio.azureedge.net/pause-amd64:3.1 --pod-manifest-path=/etc/kubernetes/manifests --register-node=true --register-with-taints=node-role.kubernetes.io/master=true:NoSchedule
k8s       36811  33970  0 22:32 pts/0    00:00:00 grep --color=auto kubelet
`

**Comment by khaldoune** [comment]
`k8s@k8s-master-35332392-0:~$ journalctl -u kubelet
Hint: You are currently not seeing messages from other users and the system.
      Users in the 'systemd-journal' group can see all messages. Pass -q to
      turn off this notice.
-- No entries --
k8s@k8s-master-35332392-0:~$
`

**Comment by idanshahar** [comment]
@khaldoune 
can you run "journalctl -u kubelet" on the master and upload the output?

**Comment by khaldoune** [comment]
@idanshahar 
Please see my last post. journalctl responds "No entries"

**Comment by CecileRobertMichon** [comment]
@khaldoune try `sudo journalctl -u kubelet`

**Comment by khaldoune** [comment]
@CecileRobertMichon @idanshahar 
Thanks.
Here you go. 
[kubelet-master.zip](https://github.com/Azure/acs-engine/files/1839468/kubelet-master.zip)

**Comment by khaldoune** [comment]
@jackfrancis

**Comment by idanshahar** [comment]
@khaldoune 
First, I suggest you to deploy a simpler api model.
Can you go over your api model and remove unnecessary configurations?  

`Mar 22 22:22:23 k8s-master-35332392-0 docker[36510]: I0322 22:22:23.305226   36534 kubelet_node_status.go:497] Using Node Hostname from cloudprovider: "k8s-master-35332392-0"
Mar 22 22:22:28 k8s-master-35332392-0 docker[36510]: W0322 22:22:28.104981   36534 cni.go:171] Unable to update cni config: No networks found in /etc/cni/net.d
Mar 22 22:22:28 k8s-master-35332392-0 docker[36510]: E0322 22:22:28.105165   36534 kubelet.go:2104] Container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized`

It seems that something goes wrong with the CNI plugin. 
@jackfrancis I think he used calico as a network policy and not azure.

**Comment by khaldoune** [comment]
@idanshahar Honestly I don't see what I can remove from my cluster definition, everything was working on 0.13, everything but the internal LB that has been fixed on k8s 1.9.6...
How can we enable more logs? @jackfrancis @CecileRobertMichon 
How to go further

**Comment by jackfrancis** [comment]
Thanks @idanshahar for helping with this thread! @khaldoune Let's do this to repro. CSE errors are a result of deployment errors, so I assume you're able to continue trying to deploy a cluster until we fix this.

So, above you pasted the api model JSON file that was generated _from_ acs-engine. Could you paste contents of the api model JSON that you inputted `into` acs-engine? (e.g., `acs-engine generate <path-to-json>`)

Additionally, try using the latest release of acs-engine to deploy your new cluster:

https://github.com/Azure/acs-engine/releases/tag/v0.14.4

**Comment by khaldoune** [comment]
@idanshahar Many thanks too
@jackfrancis Thanks for your help, and here we go:

As a temporary workaround, I've made some changes to 0.13 in order to add support for k8s 1.9.6 (I need this version to fix the ILB issue), I will be able to test it tomorrow.

Here is the cluster definition:

`{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.9",
      "orchestratorVersion": "1.9.6",
      "kubernetesConfig": {
        "networkPolicy": "calico",
        "etcdDiskSizeGB": "16",
        "enableAggregatedAPIs": true,
        "enablePodSecurityPolicy": true,
        "EnableRbac": true,
        "clusterSubnet": "10.10.0.0/16",
        "serviceCidr": "192.168.1.0/24",
        "dnsServiceIP": "192.168.1.10",
         "kubeletConfig": {
            "--event-qps": "0",
            "--non-masquerade-cidr": "10.10.0.0/16",
            "--authentication-token-webhook": "true"
         },
         "controllerManagerConfig": {
            "--address": "0.0.0.0",
            "--profiling": "false",
            "--terminated-pod-gc-threshold": "100",
            "--node-cidr-mask-size": "27",
            "--node-monitor-grace-period": "40s",
            "--pod-eviction-timeout": "60s",
            "--horizontal-pod-autoscaler-use-rest-clients": "true"
         },
         "cloudControllerManagerConfig": {
            "--profiling": "false"
         },
         "apiServerConfig": {
            "--profiling": "false",
            "--repair-malformed-updates": "false",
            "--endpoint-reconciler-type": "lease"
         },
         "addons": [
           {
             "name": "tiller",
             "enabled": false
           }
         ]
      }
    },
    "masterProfile": {
      "dnsPrefix": "k8s-noprd",
      "vnetCidr": "198.18.184.0/22",
      "count": 5,
      "vmSize": "Standard_D2_v2",
      "OSDiskSizeGB": 80,
      "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/master_subnet",
      "firstConsecutiveStaticIP": "198.18.187.230",
        "preProvisionExtension": {
          "name": "setup"
        }
    },
    "agentPoolProfiles": [
      {
        "name": "wbronze",
        "count": 1,
        "vmSize": "Standard_D2_v2",
        "OSDiskSizeGB": 80,
        "availabilityProfile": "AvailabilitySet",
        "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/worker_subnet",
        "diskSizesGB": [ 50 ],
        "StorageProfile": "ManagedDisks",
        "preProvisionExtension": {
          "name": "setup_node"
        }
      },
      {
        "name": "wsilver",
        "count": 1,
        "vmSize": "Standard_D2_v2",
        "OSDiskSizeGB": 80,
        "availabilityProfile": "AvailabilitySet",
        "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/worker_subnet",
        "diskSizesGB": [ 50 ],
        "StorageProfile": "ManagedDisks",
        "preProvisionExtension": {
          "name": "setup_node"
        }
      },
      {
        "name": "wgold",
        "count": 1,
        "vmSize": "Standard_D2_v2",
        "OSDiskSizeGB": 80,
        "availabilityProfile": "AvailabilitySet",
        "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/worker_subnet",
        "diskSizesGB": [ 50 ],
        "StorageProfile": "ManagedDisks",
        "preProvisionExtension": {
          "name": "setup_node"
        }
      },
      {
        "name": "wplatin",
        "count": 1,
        "vmSize": "Standard_D2_v2",
        "OSDiskSizeGB": 80,
        "availabilityProfile": "AvailabilitySet",
        "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/worker_subnet",
        "diskSizesGB": [ 50 ],
        "StorageProfile": "ManagedDisks",
        "preProvisionExtension": {
          "name": "setup_node"
        }
      },
      {
        "name": "wdiamond",
        "count": 1,
        "vmSize": "Standard_D2_v2",
        "OSDiskSizeGB": 80,
        "availabilityProfile": "AvailabilitySet",
        "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/worker_subnet",
        "diskSizesGB": [ 50 ],
        "StorageProfile": "ManagedDisks",
        "preProvisionExtension": {
          "name": "setup_node"
        }
      }
    ],
    "linuxProfile": {
      "adminUsername": "k8s",
      "ssh": {
        "publicKeys": [
          {
            "keyData": "xxxxxxxxxxxxxxxxxx"
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "clientId": "xxxxxxxxxxxxxxxxxx",
      "secret": "xxxxxxxxxxxxxxxxxx"
    },
    "extensionProfiles": [
        {
            "name": "setup_node",
            "version": "v1",
            "script": "setup.sh",
            "rootURL": "https://gitlab.com/octo-carrefour-k8s/acs-extensions/raw/master/",
            "extensionParameters": "198.18.192.4 k8s-noprd.xpod.carrefour.com"
        },
        {
            "name": "setup",
            "version": "v1",
            "script": "setup.sh",
            "rootURL": "https://gitlab.com/octo-carrefour-k8s/acs-extensions/raw/master/",
            "extensionParameters": "198.18.192.4 k8s-noprd.xpod.carrefour.com"
        }
      ]
  }
}
`

If you need to reproduce in the same conditions, then you may need to install a DNS and associate it with your k8s vnet (198.18.192.4 in my cluster definition).

Thanks.

**Comment by CecileRobertMichon** [comment]
@khaldoune I was able to repro just now. The deployment fails because all the nodes (masters + agents)  are `NotReady`. Trying to find the cause. I will keep you posted.

**Comment by khaldoune** [comment]
@CecileRobertMichon Good news thanks. 
I will send cluster-provision and cloud-init-output logs as soon as I have access to these VMs.

**Comment by CecileRobertMichon** [comment]
@khaldoune I think I have the cause. There seems to be a regression with Calico. I'm trying to find out which commit introduced the regression. In the meantime, if it's an option for you, the same apimodel will work if you remove the line `"networkPolicy": "calico", `.

**Comment by khaldoune** [comment]
@CecileRobertMichon I've already tried to go from 0.14.0 and replace calico 0.7 with 0.1 (I've updated the tgz url), the provisioning had failed. 

I've also seen something strange in calico's manifest: cniversion:0.1 instead of 0.7, changing it to 0.7 did not change anything.

I hope it helps.

**Comment by CecileRobertMichon** [comment]
+@dtzar who is working on a PR to upgrade Calico (#2521 ) and might be able to provide insight on the above.

**Comment by CecileRobertMichon** [comment]
To clarify, the regression is not a general calico regression as deployments using `"networkPolicy": "calico"` in our regression tests are succeeding, but rather a regression with this particular apimodel, most likely involving another custom property that became incompatible with calico in v0.14.0.

**Comment by dtzar** [comment]
The version of calico in the master branch being deployed is quite old 2.6.3 [See releases](https://github.com/projectcalico/calico/releases).  Could you check to see if the latest version in my PR referenced above resolves your problem?

As mentioned - the script extension will fail if the nodes are not ready.  The `calico-node` daemonset needs to be operational in order for the scripts to finish/pass and nodes to be ready.  

I haven't done any digging, but one suspect is that the `kubeClusterCidr` is different than your listed `vnetCidr` in your apimodel.  The `kubeClusterCidr` is the value used in the [calico network configuration here](https://github.com/Azure/acs-engine/blob/549d57231ec4fc7663e354d948f7c1481fcdde05/parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml#L157).

**Comment by CecileRobertMichon** [comment]
Thanks @dtzar! @khaldoune can you try changing the value of "clusterSubnet" to match the value of "vnetCidr"?

**Comment by dtzar** [comment]
To clarify and record, `clusterSubnet` translates into `kubeClusterCidr` in [engine.go Line](https://github.com/Azure/acs-engine/blob/549d57231ec4fc7663e354d948f7c1481fcdde05/pkg/acsengine/engine.go#L649) 649 :)

**Comment by dtzar** [comment]
@khaldoune It would be good to understand what's going on with your network topology/configuration.  Per your above configuration, I see:
"dnsServiceIP": "192.168.1.10"
"serviceCidr": "192.168.1.0/24"
"clusterSubnet": "10.10.0.0/16"
Master - "vnetCidr": "198.18.184.0/22"

**Comment by khaldoune** [comment]
Hi,

Thanks all for your assistance, I was out of the office yesterday...

@CecileRobertMichon , we need Calico because we are using it for project/namespaces/network isolation.

@dtzar : I had workarround the issue 2202 by disabling the Encryption at Rest.
 
If my understanding is good, whe should have clusterSubnet=kubeClusterCidr=clusterSubnet= PODs CIDR.

From a design point of view, the pods's cidr should be private (not directly addressable from outside the k8s cluster), and thus, we should be able to use something else than the masters and workers cidrs as a pod CIDR. That's what I'm trying to achieve. 

In Azure, a VNET can have several address spaces, so if we reed here: [https://github.com/Azure/acs-engine/blob/master/docs/kubernetes/features.md#feat-custom-vnet](https://github.com/Azure/acs-engine/blob/master/docs/kubernetes/features.md#feat-custom-vnet)

_"Additionally, to prevent source address NAT'ing within the VNET, we assign to the vnetCidr property in masterProfile the CIDR block that represents the usable address space in the existing VNET"_

I understand that I just need to add another address space (10.10.0.0/16) to my k8s VNET (198.18.184.0/22) and the magic should happen.

I've just deployed with success a K8S 1.9.6 using a modified version of the acs 0.13.1

$ k get nodes
NAME                      STATUS    ROLES     AGE       VERSION
k8s-master-35332392-0     Ready     master    2h        v1.9.6
k8s-master-35332392-1     Ready     master    2h        v1.9.6
k8s-master-35332392-2     Ready     master    2h        v1.9.6
k8s-master-35332392-3     Ready     master    2h        v1.9.6
k8s-master-35332392-4     Ready     master    2h        v1.9.6
k8s-wbronze-35332392-0    Ready     agent     2h        v1.9.6
k8s-wdiamond-35332392-0   Ready     agent     2h        v1.9.6
k8s-wgold-35332392-0      Ready     agent     2h        v1.9.6
k8s-wplatin-35332392-0    Ready     agent     2h        v1.9.6
k8s-wsilver-35332392-0    Ready     agent     2h        v1.9.6

122/125 of the Sonobuoy tests on this cluster are passing (I will analyse 3 KOs later)

Here is the complete configuration of the subnet: 
[vpod0a-k8s-prd-1-vnet.zip](https://github.com/Azure/acs-engine/files/1856974/vpod0a-k8s-prd-1-vnet.zip)

An exerpt: 
`"properties": {
                "provisioningState": "Succeeded",
                "resourceGuid": "xxxxxxxxxxxxxxxxxx",
                "addressSpace": {
                    "addressPrefixes": [
                        "198.18.184.0/21",
                        "172.16.0.0/16"
                    ]
                },`

I've replaced 10.0.0.0/16 with 172.16.0.0/16 because the first one is used.

As you can see, I've 2 address spaces in my VNET.

@CecileRobertMichon  I've also been able to deploy a k8s 1.9.3 cluster using acs0.13.1

k get nodes
NAME                      STATUS    ROLES     AGE       VERSION
k8s-master-35332392-0     Ready     master    6m        v1.9.3
k8s-master-35332392-1     Ready     master    6m        v1.9.3
k8s-master-35332392-2     Ready     master    7m        v1.9.3
k8s-master-35332392-3     Ready     master    6m        v1.9.3
k8s-master-35332392-4     Ready     master    6m        v1.9.3
k8s-wbronze-35332392-0    Ready     agent     7m        v1.9.3
k8s-wdiamond-35332392-0   Ready     agent     7m        v1.9.3
k8s-wgold-35332392-0      Ready     agent     7m        v1.9.3
k8s-wplatin-35332392-0    Ready     agent     7m        v1.9.3
k8s-wsilver-35332392-0    Ready     agent     7m        v1.9.3

I will try to provision a cluster with the PR2551. I will keep you updated.

**Comment by khaldoune** [comment]
@CecileRobertMichon @dtzar 

The provisioning with PR2521 has failed.

$ acs-engine version
Version: canary
GitCommit: ae4590a
GitTreeState: clean

Here are logs (provision, cloud init, cloud init output):
[azurelogs.zip](https://github.com/Azure/acs-engine/files/1857198/azurelogs.zip)

Thanks

**Comment by khaldoune** [comment]
@jackfrancis @CecileRobertMichon @dtzar 

The deployment fails even with the acs **0.14.5** and with a routable cidr for PODs (in the vnet address space):

vnet_prefix: 198.18.184.0/21
vnet_master_subnet: 198.18.190.0/24
vnet_worker_subnet: 198.18.189.0/24
vnet_master_first_ip: 198.18.190.50
k8s_pod_cidr: 198.18.184.0/22
k8s_service_cidr: 198.18.188.0/23
k8s_dns_service: 198.18.188.10

Here is the cluster definition:

{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.9",
      "orchestratorVersion": "1.9.6",
      "kubernetesConfig": {
        "networkPolicy": "calico",
        "etcdDiskSizeGB": "16",
        "enableAggregatedAPIs": true,
        "enablePodSecurityPolicy": true,
        "EnableRbac": true,
        "clusterSubnet": "198.18.184.0/22",
        "serviceCidr": "198.18.188.0/23",
        "dnsServiceIP": "198.18.188.10",
         "kubeletConfig": {
            "--event-qps": "0",
            "--non-masquerade-cidr": "198.18.184.0/22",
            "--authentication-token-webhook": "true"
         },
         "controllerManagerConfig": {
            "--address": "0.0.0.0",
            "--profiling": "false",
            "--terminated-pod-gc-threshold": "100",
            "--node-cidr-mask-size": "27",
            "--node-monitor-grace-period": "40s",
            "--pod-eviction-timeout": "60s",
            "--horizontal-pod-autoscaler-use-rest-clients": "true"
         },
         "cloudControllerManagerConfig": {
            "--profiling": "false"
         },
         "apiServerConfig": {
            "--profiling": "false",
            "--repair-malformed-updates": "false",
            "--endpoint-reconciler-type": "lease"
         },
         "addons": [
           {
             "name": "tiller",
             "enabled": false
           }
         ]
      }
    },
    "masterProfile": {
      "dnsPrefix": "k8s-noprd",
      "vnetCidr": "198.18.190.0/24",
      "count": 5,
      "vmSize": "Standard_D2_v2",
      "OSDiskSizeGB": 80,
      "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/master_subnet",
      "firstConsecutiveStaticIP": "198.18.190.50",
        "preProvisionExtension": {
          "name": "setup"
        }
    },
    "agentPoolProfiles": [
      {
        "name": "wbronze",
        "count": 1,
        "vmSize": "Standard_D2_v2",
        "OSDiskSizeGB": 80,
        "availabilityProfile": "AvailabilitySet",
        "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/worker_subnet",
        "diskSizesGB": [ 50 ],
        "StorageProfile": "ManagedDisks",
        "preProvisionExtension": {
          "name": "setup_node"
        }
      },
      {
        "name": "wsilver",
        "count": 1,
        "vmSize": "Standard_D2_v2",
        "OSDiskSizeGB": 80,
        "availabilityProfile": "AvailabilitySet",
        "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/worker_subnet",
        "diskSizesGB": [ 50 ],
        "StorageProfile": "ManagedDisks",
        "preProvisionExtension": {
          "name": "setup_node"
        }
      },
      {
        "name": "wgold",
        "count": 1,
        "vmSize": "Standard_D2_v2",
        "OSDiskSizeGB": 80,
        "availabilityProfile": "AvailabilitySet",
        "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/worker_subnet",
        "diskSizesGB": [ 50 ],
        "StorageProfile": "ManagedDisks",
        "preProvisionExtension": {
          "name": "setup_node"
        }
      },
      {
        "name": "wplatin",
        "count": 1,
        "vmSize": "Standard_D2_v2",
        "OSDiskSizeGB": 80,
        "availabilityProfile": "AvailabilitySet",
        "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/worker_subnet",
        "diskSizesGB": [ 50 ],
        "StorageProfile": "ManagedDisks",
        "preProvisionExtension": {
          "name": "setup_node"
        }
      },
      {
        "name": "wdiamond",
        "count": 1,
        "vmSize": "Standard_D2_v2",
        "OSDiskSizeGB": 80,
        "availabilityProfile": "AvailabilitySet",
        "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxx/resourceGroups/vpod0a-apps-prd-rg/providers/Microsoft.Network/virtualNetworks/vpod0a-k8s-prd-1-vnet/subnets/worker_subnet",
        "diskSizesGB": [ 50 ],
        "StorageProfile": "ManagedDisks",
        "preProvisionExtension": {
          "name": "setup_node"
        }
      }
    ],
    "linuxProfile": {
      "adminUsername": "k8s",
      "ssh": {
        "publicKeys": [
          {
            "keyData": "ssh-rsa xxxxxxxxxxxxxxxxxx"
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "clientId": "xxxxxxxxxxxxxxxxxx",
      "secret": "xxxxxxxxxxxxxxxxxx"
    },
    "extensionProfiles": [
        {
            "name": "setup_node",
            "version": "v1",
            "script": "setup.sh",
            "rootURL": "https://gitlab.com/octo-carrefour-k8s/acs-extensions/raw/master/",
            "extensionParameters": "198.18.192.4 k8s-noprd.xpod.carrefour.com"
        },
        {
            "name": "setup",
            "version": "v1",
            "script": "setup.sh",
            "rootURL": "https://gitlab.com/octo-carrefour-k8s/acs-extensions/raw/master/",
            "extensionParameters": "198.18.192.4 k8s-noprd.xpod.carrefour.com"
        }
      ]
  }
}
~

**Comment by khaldoune** [comment]
@jackfrancis @idanshahar @CecileRobertMichon @dtzar 

My /etc/cni is empty. Where/When its content gets created by acs-engine?

Thanks.

**Comment by CecileRobertMichon** [comment]
@khaldoune /etc/cni should contain net.d: 
https://github.com/Azure/acs-engine/blob/b38a17074e24af2af53f8649a001eaabc1575dc2/parts/k8s/kubernetesmastercustomscript.sh#L229
https://github.com/Azure/acs-engine/blob/b38a17074e24af2af53f8649a001eaabc1575dc2/parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml#L212
Since you were able to deploy the same api model with two vnets in v13.1 and see ready nodes this might be a regression. I suspect it could be linked to issue #2476. Could you please open a new issue since I think we are outside the scope of this current issue for better tracking of the bug/fix? Thank you for your patience, let's get this resolved asap! cc @jackfrancis

**Comment by khaldoune** [comment]
@CecileRobertMichon @jackfrancis
 
Provisioning using Azure CNI instead of Calico with acs 0.14.5 works fine.

Provisioning with Calico and a single subnet for both masters and workers fails.

I've also double-checked weither or not the Encryption At Rest has been enabled by default in 0.14.5, it is not.

I've just created a new issue: #2607

Thanks for your help.

**Comment by marty2bell** [comment]
I got this error yesterday using acs-engine 15.2 with the distro set to coreos.  Removing this from the template and reverting to ubuntu mitigated the issue, but means we can't provision coreOS vms.

Marty

**Comment by rocketraman** [comment]
I just upgraded a cluster from 1.7.5 to 1.8.10 via acs-engine 0.15.2 and ran into this issue. The cluster uses Azure CNI and Ubuntu.

The resource group Deployment is still showing the Failure if more details are needed.

Ignoring the error, and resuming the upgrade seems to have worked fine, but the cse0 extension on the master VM is still showing status "Provisioning failed". I don't know what the implications of this are, but as I said, everything seems to be working.

**Comment by BrendanThompson** [comment]
I am seeing this same issue with the following:

|Application|Version|
|---|---|
|`acs-engine`|v0.16.0|
|`k8s`|1.10|

The cluster is trying to use Ubuntu with Azure CNI

**Comment by CecileRobertMichon** [comment]
@rocketraman and @BrendanThompson please share the apimodel you used to generate the template/deploy the cluster as well as the exact error message (what was the error code?).

**Comment by rocketraman** [comment]
Here is my API model, with private information elided:

[apimodel.json](https://github.com/Azure/acs-engine/files/1943992/apimodel.json.txt)

Here is the error (operation status was "Conflict", Provisioning state is "Failed"):

```
{
  "status": "Failed",
  "error": {
    "code": "ResourceDeploymentFailure",
    "message": "The resource operation completed with terminal provisioning state 'Failed'.",
    "details": [
      {
        "code": "VMExtensionProvisioningError",
        "message": "VM has reported a failure when processing extension 'cse0'. Error message: \"Enable failed: failed to execute command: command terminated with exit status=3\n[stdout]\n\n[stderr]\n\"."
      }
    ]
  }
}
```

Same exact error on two different clusters.

**Comment by rocketraman** [comment]
@CecileRobertMichon I think I understand what happened in my case.

Looking at `/var/log/azure/cluster-provision.log`, it looks like it failed because it couldn't connect to etcd. This is probably because of some customizations I had made to the pre-upgrade cluster due to some other weirdness [1]. This prevented the upgraded cluster etcd from starting up and the cse0 script to fail.

[1] In my previous cluster, I was experiencing an issue in which etcd wasn't starting up because it was choking on the `lost+found` directory in `/var/lib/etcddisk/`. To fix this, I had manually moved the etcd data directory on the older cluster into a subdirectory. Thankfully, I don't have this issue with etcd on my current cluster.

**Comment by hensilva** [comment]
I'm facing similar issues today. 1.9 or 1.10 with ACS 0.16.1

**Comment by hmarcelodn** [comment]
Same issue with k8s 1.6.6 and acs-engine 1.16.2

**Comment by CecileRobertMichon** [comment]
For everyone here, https://github.com/Azure/acs-engine/blob/master/docs/kubernetes/troubleshooting.md#vmextensionprovisioningerror-or-vmextensionprovisioningtimeout has been added to help troubleshoot VM extension errors. Please follow the instructions if you encounter one of those.

**Comment by Navlesh** [comment]
@CecileRobertMichon I too face #vmextensionprovisioningtimeout error all the time when i have 3 masters.
I am using ACS v0.18.6
following is the sample input file.
{
  "apiVersion": "vlabs",
  "properties": {
    "orchestratorProfile": {
      "orchestratorType": "Kubernetes",
      "orchestratorRelease": "1.9",
      "kubernetesConfig": {	  
		"privateCluster": {
          "enabled": true		  
		}
      }
    },
    "masterProfile": {
      "count": 3,
      "dnsPrefix": "egsms",
      "vmSize": "Standard_D2s_v3",
      "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxxxxxxxx/resourceGroups/MyRG/providers/Microsoft.Network/virtualNetworks/vnet/subnets/frontend",
		"firstConsecutiveStaticIP": "10.0.0.45",
		"vnetCidr": "10.0.0.0/24"
    },
    "agentPoolProfiles": [
      {
        "name": "egsagent",
        "count": 1,
        "vmSize": "Standard_D2s_v3",
        "vnetSubnetId": "/subscriptions/xxxxxxxxxxxxxxxxxxxxxxxxxx/resourceGroups/MyRG/providers/Microsoft.Network/virtualNetworks/vnet/subnets/frontend",
        "availabilityProfile": "AvailabilitySet"
      }
    ],
    "linuxProfile": {
      "adminUsername": "useradmin",
      "ssh": {
        "publicKeys": [
          {
            "keyData": "ssh-rsa xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
          }
        ]
      }
    },
    "servicePrincipalProfile": {
      "clientId": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
      "secret": "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx="
    }
  }
}

**Comment by CecileRobertMichon** [comment]
@Navlesh please take a look at  https://github.com/Azure/acs-engine/blob/master/docs/kubernetes/troubleshooting.md#vmextensionprovisioningerror-or-vmextensionprovisioningtimeout if you haven't already and open a new issue with title "CSE error: exit code <INSERT_YOUR_EXIT_CODE>" and include the following in the description:
- The apimodel json used to deploy the cluster (aka your cluster config). Please make sure you remove all secrets and keys before posting it on GitHub (what you pasted above)
- The output of kubectl get nodes
- The content of /var/log/azure/cluster-provision.log and /var/log/cloud-init-output.log
